presto-cassandra/src/test/java/com/facebook/presto/cassandra/TestCassandraIntegrationSmokeTest.java

```diff
@@ -195,6 +203,144 @@ public void testClusteringKeyOnlyPushdown()
+        // that have differences only in letters case.
+        session.execute("CREATE KEYSPACE \"KeYsPaCe_3\" WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor': 1}");
+        session.execute("CREATE KEYSPACE \"kEySpAcE_3\" WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor': 1}");
+
+        // Although in Presto all the schema and table names are always displayed as lowercase
+        assertContainsEventually(() -> execute("SHOW SCHEMAS FROM cassandra"), resultBuilder(getSession(), createUnboundedVarcharType())
+                .row("keyspace_3")
+                .row("keyspace_3")
+                .build(), new Duration(1, MINUTES));
+
+        // There is no way to figure out what the exactly keyspace we want to retrieve tables from
+        assertQueryFails("SHOW TABLES FROM cassandra.keyspace_3",
+                "More than one keyspace has been found for the case insensitive schema name: keyspace_3 -> \\(KeYsPaCe_3, kEySpAcE_3\\)");
+
+        session.execute("DROP KEYSPACE \"KeYsPaCe_3\"");
+        session.execute("DROP KEYSPACE \"kEySpAcE_3\"");
+    }
+
+    @Test
+    public void testTableNameAmbiguity()
+            throws Exception
+    {
+        session.execute("CREATE KEYSPACE keyspace_4 WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor': 1}");
+        assertContainsEventually(() -> execute("SHOW SCHEMAS FROM cassandra"), resultBuilder(getSession(), createUnboundedVarcharType())
+                .row("keyspace_4")
+                .build(), new Duration(1, MINUTES));
+
+        // Identifiers enclosed in double quotes are stored in Cassandra verbatim. It is possible to create 2 tables with names
+        // that have differences only in letters case.
+        session.execute("CREATE TABLE keyspace_4.\"TaBlE_4\" (column_4 bigint PRIMARY KEY)");
+        session.execute("CREATE TABLE keyspace_4.\"tAbLe_4\" (column_4 bigint PRIMARY KEY)");
+
+        // Although in Presto all the schema and table names are always displayed as lowercase
+        assertContainsEventually(() -> execute("SHOW TABLES FROM cassandra.keyspace_4"), resultBuilder(getSession(), createUnboundedVarcharType())
+                .row("table_4")
+                .row("table_4")
+                .build(), new Duration(1, MINUTES));
+
+        // There is no way to figure out what the exactly table is being queried
+        assertQueryFails("SHOW COLUMNS FROM cassandra.keyspace_4.table_4",
+                "More than one table has been found for the case insensitive table name: table_4 -> \\(TaBlE_4, tAbLe_4\\)");
+        assertQueryFails("SELECT * FROM cassandra.keyspace_4.table_4",
+                "More than one table has been found for the case insensitive table name: table_4 -> \\(TaBlE_4, tAbLe_4\\)");
+
+        session.execute("DROP KEYSPACE keyspace_4");
+    }
+
+    @Test
+    public void testColumnNameAmbiguity()
```

Comment:
Please add tests for CTAS as well -- there's a slightly different codepath there